src/Arcus.WebApi.Unit/Security/Authentication/SharedAccessKeyAuthenticationAttributeTests.cs

```diff
@@ -0,0 +1,150 @@
+           string secretValue = $"secret-{Guid.NewGuid()}";
+            _testServer.AddService<ISecretProvider>(new InMemorySecretProvider((SecretName, secretValue)));
+
+            // Act
+            using (HttpResponseMessage response = await SendAuthorizedHttpRequestWithHeader(HeaderName, $"something else then {nameof(secretValue)}"))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            }
+        }
+
+        [Fact]
+        public async Task AuthorizedRoute_WithSharedAccessKey_ShouldFailWithUnauthorized_WhenHeaderIsNotPresent()
+        {
+            // Arrange
+            string secretValue = $"secret-{Guid.NewGuid()}";
+            _testServer.AddService<ISecretProvider>(new InMemorySecretProvider((SecretName, secretValue)));
+
+            // Act
+            using (HttpResponseMessage response = await SendAuthorizedHttpRequestWithHeader($"something-else-then-{nameof(HeaderName)}", "some header value"))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            }
+        }
+
+        [Fact]
+        public async Task AuthorizedRoute_WithSharedAccessKey_ShouldFailWithUnauthorized_WhenAnyHeaderValue()
+        {
+            // Arrange
+            string secretValue = $"secret-{Guid.NewGuid()}";
+            _testServer.AddService<ISecretProvider>(new InMemorySecretProvider((SecretName, secretValue)));
+
+            // Act
+            using (HttpResponseMessage response = 
+                await SendAuthorizedHttpRequestWithHeader(HeaderName, new [] { secretValue, $"second header value other then {nameof(secretValue)}" }))
+            {
+                // Assert
+                Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            }
+        }
+
+        private Task<HttpResponseMessage> SendAuthorizedHttpRequestWithHeader(string headerName, string headerValue)
+        {
+            return SendAuthorizedHttpRequestWithHeader(headerName, new[] { headerValue });
+        }
+
+        private async Task<HttpResponseMessage> SendAuthorizedHttpRequestWithHeader(string headerName, IEnumerable<string> headerValues)
+        {
+            using (HttpClient client = _testServer.CreateClient())
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, AuthorizedRoute)
+                {
+                    Headers = { { headerName, headerValues } }
+                };
+
+                return await client.SendAsync(request);
+            }
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            _testServer.Dispose();
+        }
+    }
+    
+    [ApiController]
+    public class SharedAccessKeyAuthenticationController : ControllerBase
```

Comment:
Let's only use one class per file

Reply:
Well, about that. Would that make the test more readable? SInce then you have to look in different files for a single test while the fixture setup is now in the same collection of tests.

I know we often have different opinions on this (both in prod- als test-code)

Reply:
But if you want this I can change it. Seperate in a `.\Fixtures\` folder?

Reply:
It's more to be consistent with all the other files/classes as they don't do that as well. Next to that it's also to make the class more discoverable.

I'll leave it up to you to decide but I'm not sure how it's directly related to the tests itself as well, but that's ok.